nixos/modules/services/networking/bee.nix

```diff
@@ -0,0 +1,166 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.bee;
+in {
+  meta = {
+    # doc = ./bee.xml;
+    maintainers = with maintainers; [ attila-lendvai ];
+  };
+
+  ### interface
+
+  options = {
+    services.bee = {
+      enable = mkEnableOption "Start up Ethereum Swarm Bee";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.bee;
+        defaultText = "pkgs.bee";
+        example = "pkgs.bee-unstable";
+        description = "The package providing the bee binary for the service.";
+      };
+
+      ### NOTE let's call the nix variables the same as bee calls them
+
+      config = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/var/lib/bee/bee.yaml";
+        #description = "The configuration file path.";
+      };
+
+      api-addr = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''HTTP API listen address (default ":1633")'';
+      };
+
+      data-dir = mkOption {
+        type = types.path;
+        default = "/var/lib/bee/";
+        #description = "The data directory.";
+      };
+
+      extraConfig = mkOption {
+        type = types.lines;
+        default = "";
+        example = ''
+          password-file: /var/lib/bee/password
+          payment-threshold: 100000
+        '';
+        description = "Additional text to be appended at the end of the default <filename>bee.yaml</filename>. Note that it's ignored when a custom config file is specified.";
+      };
+    };
+  };
+
+  ### implementation
+
+#  config = mkIf cfg.enable {
+  config = {
+    systemd.services.bee =
+      let
+        defaultConfigFile = pkgs.writeText "bee.yaml" ''
+## Bee configuration - https://swarm-gateways.net/bzz:/docs.swarm.eth/docs/installation/configuration
+##
+## Bee under NixOS is configured using Nix . This is a
+## generated file that will be overwritten, DO NOT EDIT.
+
+## HTTP API listen address (default ":1633")
+# api-addr: :1633
```

Comment:
If this is a default value then it doesn't seem to actually be needed here. Right?

Reply:
the service file part is very much WIP. the reason i have pushed this draft PR is to ask for help on how i can try/test it, because the relevant part seems to be bitrotten in the manual.